astropy/io/votable/tree.py

```diff
@@ -1902,37 +1905,41 @@ def ID(self, ID):
     def system(self):
         """Specifies the type of coordinate system.
 
-        Valid choices are:
-
-          'eq_FK4', 'eq_FK5', 'ICRS', 'ecl_FK4', 'ecl_FK5', 'galactic',
-          'supergalactic', 'xy', 'barycentric', or 'geo_app'
+        Valid choices are given by `~astropy.io.votable.tree.CooSys.reference_frames`
         """
         return self._system
 
     @system.setter
     def system(self, system):
-        if system not in (
-            "eq_FK4",
-            "eq_FK5",
-            "ICRS",
-            "ecl_FK4",
-            "ecl_FK5",
-            "galactic",
-            "supergalactic",
-            "xy",
-            "barycentric",
-            "geo_app",
-        ):
-            if not self._config.get("version_1_5_or_later"):
-                # Starting in v1.5, system values come from the IVOA refframe vocabulary
-                # (http://www.ivoa.net/rdf/refframe).  For now we are not checking those values.
-                warn_or_raise(E16, E16, system, self._config, self._pos)
+        if system not in self.reference_frames:
+            warn_or_raise(E16, E16, system, self._config, self._pos)
         self._system = system
 
     @system.deleter
     def system(self):
         self._system = None
 
+    @property
+    def reference_frames(self):
+        """The list of reference frames recognized in the IVOA vocabulary.
+
+        This is described at http://www.ivoa.net/rdf/refframe
+
+        Returns
+        -------
+        set[str]
+            The labels of the IVOA reference frames.
+        """
+        # since VOTable version 1.5, the 'system' in COOSYS follow the RDF vocabulary
+        # for reference frames. If this is updated upstream, the json version can be
+        # downloaded at the bottom of the page and replaced here.
+        if self._reference_frames is None:
+            with open(
+                Path(__file__).parent / "data/ivoa-vocalubary_refframe-v20220222.json"
```

Comment:
question for @pllim  if this is the preferred way or to use one of the `get_pkg...` utilities? I would lean towards the latter, but this works, too.

Reply:
The other alternative is to hardcode the list of valid reference frames with a clear comment explaining how to check for updates of the IVOA vocabulary. I'm really not sure that the file is necessary.

Reply:
Yet an other option is to cache the vocabulary like Markus did there https://github.com/astropy/pyvo/blob/main/pyvo/utils/vocabularies.py

Reply:
Done :) (I just did not know about this utility)